homeassistant/components/dunehd/media_player.py

```diff
@@ -126,6 +132,30 @@ def media_pause(self) -> None:
         """Pause media player."""
         self._state = self._player.pause()
 
+    async def async_play_media(self,  media_type: MediaType | str, media_id: str, **kwargs: Any) -> None:
+        """Play media from a URL or file."""
+        # Handle media_source
+        if media_source.is_media_source_id(media_id):
+            sourced_media = await media_source.async_resolve_media(
+                self.hass, media_id, self.entity_id
+            )
+            media_id = sourced_media.url
+
+        # If media ID is a relative URL, we serve it from HA.
+        media_id = async_process_play_media_url(
+            self.hass, media_id
+        )
+
+        self._state = await self.hass.async_add_executor_job(self._player.launch_media_url,media_id)
+
+    async def async_browse_media(
+        self,
+        media_content_type: MediaType | str | None = None,
+        media_content_id: str | None = None,
+    ) -> BrowseMedia:
+        """Implement the websocket media browsing helper."""
+        return await media_source.async_browse_media(self.hass, media_content_id)
```

Comment:
Can dunehd play just about any kind of local media?

Reply:
OK, I guess it's OK to not add any filtering then.

Reply:
Video and audio files work. Image files do not work currently, but that seems to be a bug in the player OS that it can't handle a dot in the URL query string.
I have reported that issue to the Dune support, no response so far.